### PR TITLE
Check if Yoast SEO metabox is hidden

### PR DESCRIPTION
**Pull request by ajuliano**
Corrects #9.

This function is the same used by Yoast SEO themselves.
https://github.com/Yoast/wordpress-seo/blob/trunk/admin/metabox/class-metabox.php#L132

### CODE REVIEW

File: setup.php [hunk 1]
```diff
@@ -82,7 +82,7 @@ public function enqueue_scripts($page_hook) {
     $current_screen = get_current_screen();
     $current_post_type = get_post_type_object($current_screen->post_type);
 
-    if($current_post_type->public === false) return;
+    if( $this->check_if_yoast_seo_is_hidden($current_post_type->name) === true ) return;
```
Comment (harryfinn) [inline setup.php]: I need to put together some code style preference for this seeing as it's now getting some external PR's 😉 

Could we format this line as (stripping unnecessary whitespace effectively):

```
 if($this->check_if_yoast_seo_is_hidden($current_post_type->name) === true) return;
```

File: setup.php [hunk 2]
```diff
@@ -92,6 +92,30 @@ public function enqueue_scripts($page_hook) {
...
+  /**
+   * Test whether Yoast SEO is hidden either by choice of the admin or because
+   * the post type is not a public post type
+   *
+   *
+   * @param  string $post_type (optional) The post type to test, defaults to the current post post_type.
+   *
+   * @return  bool        Whether or not the yoast seo metabox should be hidden
+   */
+  function check_if_yoast_seo_is_hidden( $post_type = null ) {
```
Comment (harryfinn) [inline setup.php]: seeing as we are only calling this method inside this class, could we declare this method as `private` and also remove the extra white spaces?

File: setup.php [hunk 3]
```diff
@@ -92,6 +92,30 @@ public function enqueue_scripts($page_hook) {
...
+    if ( ! isset( $post_type ) && ( isset( $GLOBALS['post'] ) && ( is_object( $GLOBALS['post'] ) && isset( $GLOBALS['post']->post_type ) ) ) ) {
```
Comment (harryfinn) [inline setup.php]: same as before, if we could remove any unnecessary whitespace (I appreciated this has just been lifted from the the plugin code). If we could do the same for lines 111 onwards too?

Perhaps its also worth storing `$GOLBALS` as a new variable here before the `if` conditions so that we aren't repeatedly checking agains the globals variable directly?

Could we also look to break this long line up too? I tend to try and aim for around 80 characters per line, a line return and 1 indent usually still looks readable

**Review by harryfinn — CHANGES_REQUESTED** [review]

File: setup.php (hunk 3 above)
Comment (ajuliano) [reply setup.php]: For sure, it's your plugin, so you do what you think is best :-)

But I do really like the whitespaces as described in the WordPress coding standards:
https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#space-usage

File: setup.php (hunk 2 above)
Comment (ajuliano) [reply setup.php]: Yep, that's better!

**Comment by ajuliano** [comment]
Thank you for the feedback. I tried to do the requested changes. I'm new with making pull requests, so give me more feedback if that's needed :-).

File: setup.php [hunk 4]
```diff
@@ -92,6 +92,34 @@ public function enqueue_scripts($page_hook) {
...
+   *
+   *
+   * @param  string $post_type (optional) The post type to test, defaults to the current post post_type.
+   *
+   * @return  bool        Whether or not the yoast seo metabox should be hidden
+   */
+  private function check_if_yoast_seo_is_hidden($post_type = null) {
+    $global_post = $GLOBALS['post'];
+
+    if (!isset($post_type) && (isset($global_post) && (is_object($global_post)
```
Comment (harryfinn) [inline setup.php]: could we also remove the space before this `if`

File: setup.php [hunk 5]
```diff
@@ -92,6 +92,34 @@ public function enqueue_scripts($page_hook) {
...
+      && isset($global_post->post_type)))) {
+      $post_type = $global_post->post_type;
+    }
+
+    if ( isset( $post_type ) ) {
```
Comment (harryfinn) [inline setup.php]: same here, remove the spaces before the `if` and around the brackets?

File: setup.php [hunk 6]
```diff
@@ -92,6 +92,34 @@ public function enqueue_scripts($page_hook) {
...
+      // Don't make static as post_types may still be added during the run.
+      $cpts    = get_post_types(array('public' => true),'names');
```
Comment (harryfinn) [inline setup.php]: could we use the newer array style syntax here `['public' => true]` instead of the `array` keyword? Could we add a space after the comma too?

**Review by harryfinn — CHANGES_REQUESTED** [review]

**Comment by harryfinn** [comment]
@ajuliano a couple more minor spacing etc we could remove before merging this - thanks for your input so far, glad to have another contributor onboard! 😄

**Comment by ajuliano** [comment]
Sorry for the late response, I've had too much work. But I hope I've made the requested changes now.

**Review by harryfinn — APPROVED** [review]

**Comment by harryfinn** [comment]
@ajuliano thanks for the amends - hope you've had a good Christmas! 🎄 🎅

**Comment by ajuliano** [comment]
Awesome! You too.
